azurelinuxagent/ga/monitor.py

```diff
@@ -153,6 +153,8 @@ def stop(self):
 
     def init_protocols(self):
         self.protocol = self.protocol_util.get_protocol()
+        # Update the GoalState first time to instantiate all required objects for the monitor thread
+        self.protocol.update_goal_state()
```

Comment:
we still have the issue that the different threads may update the files on disk... it would only affect debugging, though. I can fix that when I change the way we archive those files.

Reply:
Sounds good, thanks!